modules/phenyl-mongodb/src/mongodb-client.js

```diff
@@ -17,14 +17,33 @@ import type {
   MultiUpdateCommand,
   DeleteCommand,
   FindOperation,
+  SimpleFindOperation,
 } from 'phenyl-interfaces'
 
 import type { MongoDbConnection } from './connection.js'
 
-function setIdTo_idInWhere(where: FindOperation): FindOperation {
+// $FlowIssue(this-is-SimpleFindOperation)
+function setIdTo_idInWhere(simpleFindOperation: SimpleFindOperation): SimpleFindOperation {
+  return assign(simpleFindOperation, { $rename: { id: '_id' } })
+}
+
+function convertDocumentPathToDotNotation(simpleFindOperation: SimpleFindOperation): SimpleFindOperation {
+  return Object.keys(simpleFindOperation).reduce((operation, srcKey) => {
+    const dstKey = srcKey.replace(/\[(\d+)]/g, (m, p) => `.${p}`)
+    operation[dstKey] = simpleFindOperation[srcKey]
+    return operation
+  }, {})
+}
+
+const WHERE_FILTERS : Array<(SimpleFindOperation) => SimpleFindOperation> = [
```

Comment:
Maybe composing functions is better, but this implementation is ok.

Reply:
96368e5 make sense?